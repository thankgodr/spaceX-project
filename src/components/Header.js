import Container from 'react-bootstrap/Container';
import Nav from 'react-bootstrap/Nav';
import Navbar from 'react-bootstrap/Navbar';
import { NavLink } from 'react-router-dom';
import NavRoutePaths from '../navigation/navroutes';
import Logo from "../images/logo.png"

const Header = () => {
    return (
        <Navbar collapseOnSelect expand="lg" bg="light" variant="light">
            <Container>
                <Navbar.Brand href="/"> <img src={Logo} className="logo"/> Space Travels' Hub</Navbar.Brand>
                <Navbar.Toggle aria-controls="responsive-navbar-nav" />
                <Navbar.Collapse id="responsive-navbar-nav">
                    <Nav className="me-auto">
                    </Nav>
                    <Nav>
                        <Nav.Link> <NavLink to={NavRoutePaths.ROCKETS}>Rockets </NavLink></Nav.Link>
                        <Nav.Link> <NavLink to={NavRoutePaths.MISSION}>Missions </NavLink></Nav.Link>
                        <Nav.Link> <NavLink to={NavRoutePaths.PROFILE}>My Profile </NavLink></Nav.Link>
                    </Nav>
                </Navbar.Collapse>
            </Container>
        </Navbar>
    )
}

export default Header