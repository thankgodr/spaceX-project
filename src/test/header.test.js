import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import Header from '../components/Header';

describe('Tesing the Header Component', () => {
  test('Test The header render correctly', () => {
    const header = renderer.create(
      <BrowserRouter>
        <Header />
      </BrowserRouter>,
    ).toJSON();
    expect(header).toMatchSnapshot();
  });

  test('navs are 3', () => {
    const { container } = render(
      <BrowserRouter>
        <Header />
      </BrowserRouter>,
    );

    const nav = container.getElementsByClassName('nav-link');
    expect(nav.length).toBe(3);
  });

  test('navs are clickable', () => {
    render(
      <BrowserRouter>
        <Header />
      </BrowserRouter>,
    );
    userEvent.click(screen.getByText('Rockets'));
    expect(screen.getByText('Rockets')).toHaveClass('active');
  });
});