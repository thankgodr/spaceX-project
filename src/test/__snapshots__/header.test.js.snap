// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tesing the Header Component Test The header render correctly 1`] = `
<nav
  className="navbar navbar-expand-lg navbar-light bg-light"
>
  <div
    className="container"
  >
    <a
      className="navbar-brand"
      href="/"
    >
      <img
        className="logo"
        src="logo.png"
      />
       Space Travels' Hub
    </a>
    <button
      aria-controls="responsive-navbar-nav"
      aria-label="Toggle navigation"
      className="navbar-toggler collapsed"
      onClick={[Function]}
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      aria-expanded={null}
      className="navbar-collapse collapse"
      id="responsive-navbar-nav"
    >
      <div
        className="me-auto navbar-nav"
        onKeyDown={[Function]}
      />
      <div
        className="navbar-nav"
        onKeyDown={[Function]}
      >
        <a
          aria-current="page"
          className="nav-link active"
          href="/"
          onClick={[Function]}
          role="button"
        >
          Rockets 
        </a>
        <a
          className="nav-link"
          href="/mission"
          onClick={[Function]}
          role="button"
          tabindex="0"
        >
          Missions 
        </a>
        <a
          className="nav-link"
          href="/myprofile"
          onClick={[Function]}
          role="button"
        >
          My Profile 
        </a>
      </div>
    </div>
  </div>
</nav>
`;
